smalltalksrc/VMMaker/StackInterpreter.class.st

```diff
@@ -5433,13 +5465,20 @@ StackInterpreter >> extStoreReceiverVariableBytecode [
 StackInterpreter >> extUnconditionalJump [
 	"242		11110010	i i i i i i i i	Jump i i i i i i i i (+ Extend B * 256, where bbbbbbbb = sddddddd, e.g. -32768 = i=0, a=0, s=1)"
 	| byte offset bcpcDelta |
+	<compilationInfo: #branch>
+	<compilationInfo: #isMapped> "because of interrupt check"
+	<compilationInfo: #v4:Long:Branch:Distance:>
 	byte := self fetchByte.
 	offset := byte + (extB << 8).
+	
+	"If backwards, we should consider the delta from before the previous extensions.
+	Consider each extension is 2 bytes."
 	bcpcDelta := offset < 0 ifTrue: [numExtB * 2] ifFalse: [0].
 	extB := 0.
 	numExtB := 0.
-	instructionPointer := instructionPointer + offset.
+	self druidIgnore: [ instructionPointer := instructionPointer + offset ].
 	self ifBackwardsCheckForEvents: offset + bcpcDelta.
+	self interpreterIgnore: [ instructionPointer := instructionPointer + offset + 1 ].
```

Comment:
I was thinking, after our talk about the IPs differences between compiler and interpreter.

Maybe we need a method
```st
self moveIP: offset + 1 andDo: [ self ifBackwardsCheckForEvents: offset + bcpcDelta ].
```

And override the definition for the compiler.